### PR TITLE
nautilus: msg/async: connection race + winner fault can leave connection stuck at replacing foreve

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/39241

### CODE REVIEW

File: src/msg/simple/Pipe.cc [hunk 1]
```diff
@@ -164,7 +164,7 @@ Pipe::Pipe(SimpleMessenger *r, int st, PipeConnection *con)
 
   randomize_out_seq();
 
-  msgr->timeout = msgr->cct->_conf->ms_tcp_read_timeout * 1000; //convert to ms
+  msgr->timeout = msgr->cct->_conf->ms_connection_idle_timeout * 1000; //convert to ms
```
Comment (pdvian) [inline src/msg/simple/Pipe.cc]: @xiexingguo Need review here. The Pipe constructor still referring to ms_tcp_read_timeout conf and has to be renamed to ms_connection_idle_timeout.

**Comment by dillaman** [comment]
Please pull in PR #28050 as well to avoid breaking RBD tests

**Comment by xiexingguo** [comment]
@pdvian Can you cherry-pick in 6b4f9729fd7cec14352b74cb3674a893159a1aa5 too? Thanks!

**Comment by pdvian** [comment]
@dillaman @xiexingguo Sure. Let me work on it.

**Comment by pdvian** [comment]
@dillaman @xiexingguo Cherry-picked #28050. Kindly review.

**Comment by yuriw** [comment]
[wip-yuri3-testing-2019-06-14-1440-nautilus](https://trello.com/c/217ldlvH/642-wip-yuri3-testing-2019-06-14-1440-nautilus)

**Pull request merged**